src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -2376,6 +2384,22 @@ void MacroAssembler::jump_cc(Condition cc, AddressLiteral dst, Register rscratch
   }
 }
 
+void MacroAssembler::cmp32_mxcsr_std(Address mxcsr_save, Register tmp, Register rscratch) {
+  ExternalAddress mxcsr_std(StubRoutines::x86::addr_mxcsr_std());
+  assert(rscratch != noreg || always_reachable(mxcsr_std), "missing");
+
+  stmxcsr(mxcsr_save);
+  movl(tmp, mxcsr_save);
+  // Mask out any pending exceptions (only check control and mask bits)
```

Comment:
This comment should go on the else path and could be changed to "Mask out status bits (only check control and mask bits)"